### PR TITLE
java-maven legacy image update

### PR DESCRIPTION
**Pull request by thomasr8**
# Description
Replaced instances of legacy images with next-gen images

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

### CODE REVIEW

File: jekyll/_cci2/language-java-maven.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ version: 2.0
 jobs:
   build:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk:stretch
```
Comment (rosieyohannan) [inline jekyll/_cci2/language-java-maven.md:62]: @thomasr8 there is no stretch variant for openjdk: https://circleci.com/developer/images/image/cimg/openjdk

@felicianotech any advice on what we should be replacing stretch variants with in examples?

File: jekyll/_cci2/language-java-maven.md [hunk 2]
```diff
@@ -96,7 +96,7 @@ jobs:
 
   build:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk.17.0.1
```
Comment (felicianotech) [inline jekyll/_cci2/language-java-maven.md:99]: Typo

`cimg/openjdk.17.0.1` should be `cimg/openjdk:17.0.1`

File: jekyll/_cci2/language-java-maven.md [hunk 3]
```diff
@@ -165,7 +165,7 @@ version: 2.0
 jobs:
   build:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk.17.0.1
```
Comment (felicianotech) [inline jekyll/_cci2/language-java-maven.md:168]: Typo

`cimg/openjdk.17.0.1` should be `cimg/openjdk:17.0.1`

File: jekyll/_cci2/language-java-maven.md [hunk 4]
```diff
@@ -235,7 +235,7 @@ jobs:
 
   build:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk.17.0.1
```
Comment (felicianotech) [inline jekyll/_cci2/language-java-maven.md:238]: Typo

`cimg/openjdk.17.0.1` should be `cimg/openjdk:17.0.1`

File: jekyll/_cci2/language-java-maven.md [hunk 5]
```diff
@@ -320,7 +320,7 @@ jobs:
   test:
     parallelism: 2
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk.17.0.1
```
Comment (felicianotech) [inline jekyll/_cci2/language-java-maven.md:323]: Typo

`cimg/openjdk.17.0.1` should be `cimg/openjdk:17.0.1`

File: jekyll/_cci2/language-java-maven.md [hunk 6]
```diff
@@ -347,7 +347,7 @@ jobs:
 
   build:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk.17.0.1
```
Comment (felicianotech) [inline jekyll/_cci2/language-java-maven.md:350]: Typo

`cimg/openjdk.17.0.1` should be `cimg/openjdk:17.0.1`

File: jekyll/_cci2/language-java-maven.md [hunk 7]
```diff
@@ -59,7 +59,7 @@ version: 2.0
 jobs:
   build:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk.17.0.1
```
[the remark this answers is not in the document]
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Fixed

File: jekyll/_cci2/language-java-maven.md [hunk 8]
```diff
@@ -86,7 +86,7 @@ version: 2.0
 jobs:
   test:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk.17.0.1
```
[the remark this answers is not in the document]
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Fixed

File: jekyll/_cci2/language-java-maven.md (hunk 2 above)
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Fixed

File: jekyll/_cci2/language-java-maven.md [hunk 9]
```diff
@@ -127,7 +127,7 @@ version: 2.0
 jobs:
   build:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: ccimg/openjdk.17.0.1
```
[the remark this answers is not in the document]
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Fixed

File: jekyll/_cci2/language-java-maven.md (hunk 3 above)
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Fixed

File: jekyll/_cci2/language-java-maven.md [hunk 10]
```diff
@@ -283,7 +283,7 @@ version: 2.0
 jobs:
   test:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk.17.0.1
```
[the remark this answers is not in the document]
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Fixed

File: jekyll/_cci2/language-java-maven.md (hunk 5 above)
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Fixed

File: jekyll/_cci2/language-java-maven.md (hunk 6 above)
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Fixed

File: jekyll/_cci2/language-java-maven.md (hunk 1 above)
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Added ":17.0" based on @felicianotech feedback.

File: jekyll/_cci2/language-java-maven.md [hunk 11]
```diff
@@ -216,7 +216,7 @@ jobs:
   test:
     parallelism: 2 # parallel containers to split the tests among
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk:stretch
```
[the remark this answers is not in the document]
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Added ":17.0" to fix this

File: jekyll/_cci2/language-java-maven.md [hunk 12]
```diff
@@ -179,7 +179,7 @@ jobs:
 
   test:
     docker:
-      - image: circleci/openjdk:stretch
+      - image: cimg/openjdk.17.0.1
```
[the remark this answers is not in the document]
Comment (thomasr8) [reply jekyll/_cci2/language-java-maven.md]: Fixed

**Review by felicianotech — APPROVED** [review]

**Pull request merged**